lib/collection/src/shards/transfer/stream_records.rs

```diff
@@ -48,10 +50,13 @@ pub(super) async fn transfer_stream_records(
             .await?;
 
         let Some(count_result) = replica_set
-            .count_local(Arc::new(CountRequestInternal {
-                filter: None,
-                exact: true,
-            }))
+            .count_local(
+                Arc::new(CountRequestInternal {
+                    filter: None,
+                    exact: true,
+                }),
+                timeout,
```

Comment:
I think we can cut the propagation here, no need to specify timeout for transfer_stream_records, I believe